sbi/inference/snle/snle_base.py

```diff
@@ -112,31 +103,16 @@ def append_simulations(
             NeuralInference object (returned so that this function is chainable).
         """
 
-        is_valid_x, num_nans, num_infs = handle_invalid_x(x, exclude_invalid_x)
-
-        x = x[is_valid_x]
-        theta = theta[is_valid_x]
-
-        # Check for problematic z-scoring
-        warn_if_zscoring_changes_data(x)
-        nle_nre_apt_msg_on_invalid_x(num_nans, num_infs, exclude_invalid_x, "SNLE")
-
-        if data_device is None:
-            data_device = self._device
-        theta, x = validate_theta_and_x(
-            theta, x, data_device=data_device, training_device=self._device
+        # pyright false positive, will be fixed with pyright 1.1.310
+        return super().append_simulations(  # type: ignore
```

Comment:
Any reason to specifically do `super().` here instead of just `self.`?

Reply:
I think this would cause a recursion call to `append_simulations` no?